plugin/pkg/admission/podresourceallocation/admission.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podresourceallocation
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"reflect"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apiserver/pkg/admission"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	api "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/apis/core/helper/qos"
+	"k8s.io/kubernetes/pkg/auth/nodeidentifier"
+	"k8s.io/kubernetes/pkg/features"
+)
+
+// PluginName is a string with the name of the plugin
+const PluginName = "PodResourceAllocation"
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return NewPodResourceAllocation(), nil
+	})
+}
+
+// Plugin contains the client used by the admission controller
+type Plugin struct {
+	*admission.Handler
+	nodeidentifier nodeidentifier.NodeIdentifier
+}
+
+var _ admission.ValidationInterface = &Plugin{}
+
+// NewPodResourceAllocation creates a new instance of the PodResourceAllocation admission controller
+func NewPodResourceAllocation() *Plugin {
+	return &Plugin{
+		Handler:        admission.NewHandler(admission.Create, admission.Update),
+		nodeidentifier: nodeidentifier.NewDefaultNodeIdentifier(),
+	}
+}
+
+// Admit may mutate the pod's ResourcesAllocated or ResizePolicy fields for compatibility with older client versions
+func (p *Plugin) Admit(ctx context.Context, attributes admission.Attributes, o admission.ObjectInterfaces) (err error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+		return nil
+	}
```

Comment:
prefer injecting the feature gates over taking a direct dependency on the global feature gate. as an example, see:

https://github.com/kubernetes/kubernetes/blob/c53ce48d48372c30053a9b67c6d1dee237b5cd69/plugin/pkg/admission/runtimeclass/admission.go#L82-L87

https://github.com/kubernetes/kubernetes/blob/c53ce48d48372c30053a9b67c6d1dee237b5cd69/plugin/pkg/admission/runtimeclass/admission.go#L99-L106

https://github.com/kubernetes/kubernetes/blob/c53ce48d48372c30053a9b67c6d1dee237b5cd69/plugin/pkg/admission/runtimeclass/admission.go#L116-L120